Allow having clickable areas one inside another

otommod opened the pull request:
Previously, if you wanted to, say, handle both scroll-up and scroll-down, you couldn't. This patch fixes that.
Some basic testing I did doesn't produce any errors.

otommod commented:
I just noticed that this has been an open request (#66) for quite some time, and that there's also a patch there...

mrshankly commented on bar.c:
```diff
@@ -1147,7 +1158,7 @@ main (int argc, char **argv)
-                                area_t *area = area_get(press_ev->event, press_ev->event_x);
+                                area_t *area = area_get(press_ev->event, press_ev->detail, press_ev->event_x);
                                 /* Respond to the click */
                                 if (area && area->button == press_ev->detail) {
```
`area->button` is already checked in the new `area_get` function, see [here](https://github.com/otommod/bar/blob/cece20b562adf3d30c2f2a071736e82371559472/bar.c#L230). Should this check be kept here?

LemonBoy replied:
Yep, the check should be omitted

otommod replied:
Indeed, fixed.

LemonBoy commented:
@mrshankly So, is this working as intended ?

mrshankly commented:
> @mrshankly So, is this working as intended ?

Yes, I think so. I've been using it for more than a week now and haven't noticed any problems.

LemonBoy commented:
Awesome, then I'll just wait for @otommod to rebase the PR to merge it :)
If you feel that something about this should be added to the documentation feel free to write a line or two!

otommod commented:
@LemonBoy rebased. I'm not too familiar with git yet, so I rebased and force pushed, I don't know if that's considered bad. Maybe I should have merged master instead?

I see you change the comments to C++-style. TBH, I prefer C-style comments, but it's your project, so I changed mine too.

andornaut commented:
It'd be great if the readme could be updated with the new syntax for multiple actions.

LemonBoy commented:
@andornaut feel free to do so, I assumed that since many people requested that feature it should be easy for them to make use of the multiple actions.

andornaut commented:
Done in #104
